### PR TITLE
[RFC] Add multi-threaded example to readme

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
I will start a parallel (😉) PR to add more detailed example and vectorised HMC functionality to the docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,33 @@ adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(0.8, integra
...
 
+### Parallel sampling 
+
+AdvancedHMC enables parallel sampling (either distributed or multi-thread) via Julia's [parallel computing functions](https://docs.julialang.org/en/v1/manual/parallel-computing/).
+It also supports vectorized sampling for static HMC and has been discussed in more detail in the documentation [here](https://turing.ml/dev/docs/library/advancedhmc/parallel_sampling).
+
+The below example utilizes the `@threads` macro to sample 4 chains across 4 threads.
+
+```julia
+#Ensure that julia was launched with appropriate number of threads
```
Comment (xukai92) [inline README.md:74]: Can we add a space after `#`?

File: README.md [hunk 2]
```diff
@@ -63,6 +63,33 @@ adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(0.8, integra
...
+
+#Number of chains to sample
+nchains = 4
+
+#Cache to store the chains
+chains = Vector{Any}(undef, nchains)
+
+#Adjust the `verbose` flag as per need
+Threads.@threads for i in 1:nchains
+  chains[i] = sample(hamiltonian, proposal, initial_θ, n_samples, adaptor, n_adapts; verbose = false)
```
Comment (xukai92) [inline README.md:85]:
```suggestion
  chains[i] = sample(hamiltonian, proposal, initial_θ, n_samples, adaptor, n_adapts; verbose=false)
```

File: README.md [hunk 3]
```diff
@@ -63,6 +63,33 @@ adaptor = StanHMCAdaptor(MassMatrixAdaptor(metric), StepSizeAdaptor(0.8, integra
...
+end
+
+#Each element of the `chains` vector contains a pair of `samples` and `stats` 
+#Concatenate the chains into a single chain
+aggregate_chain vcat([chain[1] for chain in chains]...)
```
Comment (xukai92) [inline README.md:90]: So this merges 4 chains of length N into a chain of length 4N, right? I'm not we want to do this here, instead, we can update the in-loop statement to
```julia
samples, stats = 
chains[i] = samples
```
And how to use the four chains is up-to users (either concatenated or treating them as 4 independent chains).

BTW you missed a `=` here.

**Comment by Vaibhavdixit02** [comment]
@xukai92 the suggestions were incorporated here

**Comment by xukai92** [comment]
Thanks! @Vaibhavdixit02

**Pull request merged**